File: common/src/main/kotlin/common/utils/CodeUtils.kt
package common.utils

//TODO Only works for codMode 8 aka interurban
fun createLineCode(codMode: String, lineCode: String): String = "${codMode}__${lineCode}___"
fun createStopCode(codMode: String, stopCode: String) = "${codMode}_${stopCode}"
fun getCodModeFromLineCode(input: String): String = input.substringBefore("__")
fun getSimpleLineCodeFromLineCode(input: String): String = input.substringAfter("__").substringBefore("___")
fun getStopCodeFromFullStopCode(input: String): String = input.substringAfter("_")
fun getCodModeFromFullStopCode(input: String): String = input.substringBefore("_")


const val metroCodMode = "4"
const val tramCodMode = "10"
const val trainCodMode = "5"
const val emtCodMode = "6"
const val busCodMode = "8"
const val urbanCodMode = "9"

File: MadridTransporte.Api/Utils/CodeUtils.cs
namespace MadridTransporte.Api.Utils;

public static class CodeUtils
{
    public const string MetroCodMode = "4";
    public const string TrainCodMode = "5";
    public const string EmtCodMode = "6";
    public const string BusCodMode = "8";
    public const string UrbanCodMode = "9";
    public const string TramCodMode = "10";

    public static string CreateLineCode(string codMode, string lineCode) => $"{codMode}__{lineCode}___";
    public static string CreateStopCode(string codMode, string stopCode) => $"{codMode}_{stopCode}";
    public static string GetCodModeFromLineCode(string input) => input.Split("__")[0];
    public static string GetSimpleLineCodeFromLineCode(string input) => input.Split("__")[1].Split("___")[0];
    public static string GetStopCodeFromFullStopCode(string input) => input[(input.IndexOf('_') + 1)..];
    public static string GetCodModeFromFullStopCode(string input) => input.Split('_')[0];
}
